### PR TITLE
Convert things to Typescript, including languageHandler

### PR DESCRIPTION
**Pull request by t3chguy**
Requires https://github.com/vector-im/riot-web/pull/14299

This was therapeutic.

### CODE REVIEW

File: src/languageHandler.tsx [hunk 1]
```diff
@@ -141,23 +155,25 @@ export function _t(text, variables, tags) {
...
     }
 
     if (tags !== undefined) {
-        const regexpMapping = {};
+        const regexpMapping: Tags = {};
         for (const tag in tags) {
             regexpMapping[`(<${tag}>(.*?)<\\/${tag}>|<${tag}>|<${tag}\\s*\\/>)`] = tags[tag];
         }
-        result = replaceByRegexes(result, regexpMapping);
+        result = replaceByRegexes(result as string, regexpMapping);
```
Comment (JorikSchellekens) [inline src/languageHandler.tsx:176]: Can we avoid this coerce?

Comment (t3chguy) [reply src/languageHandler.tsx:176]: Yes, but it becomes uglier IMO as we have to split the variable `result` into its type constituents, e.g:

```javascript
    let stringResult: string;
    if (variables !== undefined) {
        const regexpMapping: IVariables = {};
        for (const variable in variables) {
            regexpMapping[`%\\(${variable}\\)s`] = variables[variable];
        }
        stringResult = replaceByRegexes(text, regexpMapping);
    }

    let reactNodeResult: React.ReactNode;
    if (tags !== undefined) {
        const regexpMapping: Tags = {};
        for (const tag in tags) {
            regexpMapping[`(<${tag}>(.*?)<\\/${tag}>|<${tag}>|<${tag}\\s*\\/>)`] = tags[tag];
        }
        reactNodeResult = replaceByRegexes(stringResult || text, regexpMapping);
    }

    return reactNodeResult || stringResult || text;
```

Comment (t3chguy) [reply src/languageHandler.tsx:176]: actually, maybe I have a nicer way

Comment (JorikSchellekens) [reply src/languageHandler.tsx:176]: I think the ugliness comes from the span wrapping behaviour but yes I agree: this is not better.

**Pull request merged**